File: 108_Convert_Sorted_Array_to_BST.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def sortedArrayToBST(self, nums: List[int]) -> TreeNode:
        def recursive(low, high):
            if low == high:
                return None
            middle = (low + high) // 2
            return TreeNode(nums[middle], recursive(low, middle), recursive(middle+1, high))
        return recursive(0, len(nums))
        

File: 108_Convert_Sorted_Array__To_BST.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public TreeNode sortedArrayToBST(int[] nums) {
        int low = 0;
        int high = nums.length - 1;
        return convertToBST(nums, low, high);
    }
    public TreeNode convertToBST(int [] nums, int low, int high){
        if (low > high)
            return null;
        int middle = (low + high) / 2;
        TreeNode node = new TreeNode(nums[middle]);
        if (low == high)
            return node;
        node.left = convertToBST(nums, low, middle-1);
        node.right = convertToBST(nums, middle+1, high);
        return node;
    }
}
